doc/rst/technotes/partialInstantiations.rst

```diff
@@ -0,0 +1,255 @@
+  proc bar(arg : R(Z=uint, ?)) { ... }
+
+Furthermore, once ``?`` appears in a type constructor call all later arguments
+in the same call must used named expressions. This restriction emphasizes that
+``?`` may not be used as a positional argument. For example:
+
+.. code-block:: chapel
+
+  record R {
+    type A = int;
+    type B = real;
+  }
+
+  // Error! second argument must use a named expression
+  // proc foo(arg : R(?, string)) { ... }
+  // instead...
+  proc foo(arg : R(?, B=string)) { ... }
+
+Detecting Partial Instantiations
+--------------------------------
+
+There are a few features that allow for the detection of partial
+instantiations.
+
+The first feature is a standard library function named :proc:`~Types.isGeneric`
+that will simply return ``true`` if its argument is a partial instantiation or
+a fully-generic type:
+
+.. code-block:: chapel
+
+  record R {
+    type T;
+    param p : int;
+  }
+
+  writeln(isGeneric(R));            // true
+  writeln(isGeneric(R(int)));       // true
+  writeln(isGeneric(R(p=42)));      // true
+  writeln(isGeneric(R(string, 5))); // false
+
+Users may also query individual fields to determine whether the field has
+been instantiated by comparing the field against ``?``:
+
+.. code-block:: chapel
+
+  record R {
+    type T;
+    param p : int;
+  }
+
+  type A = R(int);
+  
+  // '?' can be used to compare against either type or param fields
+  writeln(A.T == ?); // false
+  writeln(A.p == ?); // true
+
+Finally, a Reflection function named :proc:`~Reflection.isFieldBound` will
+return true if the given name of the field in the provided type has been
+instantiated:
+
+.. code-block:: chapel
+
+  use Reflection;
+
+  record R {
+    type T;
+    type U;
+    param p : int;
+  }
+
+  proc printInstantiated(type T) {
+    writeln("type = ", T:string);
+    for param i in 1..numFields(T) {
+      param name : string = getFieldName(T, i);
+      writeln("  field ", name, " = ", isFieldBound(T, name));
+    }
+  }
+
+  printInstantiated(R);
+  printInstantiated(R(U=real));
+
+
+This program outputs:
+
+.. code-block:: text
+
+  type = R
+    field T = false
+    field U = false
+    field p = false
+  type = R(U=real(64))
+    field T = false
+    field U = true
+    field p = false
+
+
+
+Passing and Returning Generic Types
+-----------------------------------
+
+As of the 1.20 release generic type expressions may be passed to and returned
+from functions. This change supports not only partial instantiations, but also
+generic management of class types.
+
+.. code-block:: chapel
+
+  class C {
+    type T;
+    param p : int;
+  }
+
+  proc defaultC() type {
+    return C(int, 5)?;
+  }
+
+  proc intC type {
+    return C(int)?;
+  }
+
+  // Declare two variables with the same instantiation of 'C' but with different management
+  var x : unmanaged defaultC();
+  var y : owned defaultC();
+
+  // Declare two varibles with different instantiations of 'C', based on a
```

Comment:
@benharsh: @cassella points out that 'variables' is mis-spelled here.

Reply:
Thanks for passing that along and fixing it.  If I'd realized the window for merging was closing today, I'd have submitted a PR yesterday instead of waiting for more typos to batch it up with.